src/brsTypes/components/RoDeviceInfo.ts

```diff
@@ -0,0 +1,687 @@
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("eng");
+        },
+    });
+
+    private timeSinceLastKeyPress = new Callable("timeSinceLastKeyPress", {
+        signature: {
+            args: [],
+            returns: ValueKind.Int32,
+        },
+        impl: (_interpreter) => {
+            return new Int32(360);
+        },
+    });
+
+    private getDrmInfo = new Callable("getDrmInfo", {
+        signature: {
+            args: [],
+            returns: ValueKind.Object,
+        },
+        impl: (_interpreter) => {
+            let result = new Array<AAMember>();
+            result.push({ name: new BrsString("playready"), value: new BrsString("tee;ss") });
+
+            return new RoAssociativeArray(result);
+        },
+    });
+
+    private getDrmInfoEx = new Callable("getDrmInfoEx", {
+        signature: {
+            args: [],
+            returns: ValueKind.Object,
+        },
+        impl: (_interpreter) => {
+            let result = new Array<AAMember>();
+            let subresult = new Array<AAMember>();
+            subresult.push({ name: new BrsString("multikey"), value: BrsBoolean.False });
+            subresult.push({ name: new BrsString("securestop"), value: BrsBoolean.True });
+            subresult.push({ name: new BrsString("tee"), value: BrsBoolean.False });
+            subresult.push({ name: new BrsString("version"), value: new Int32(2) });
+
+            result.push({
+                name: new BrsString("playready"),
+                value: new RoAssociativeArray(subresult),
+            });
+
+            return new RoAssociativeArray(result);
+        },
+    });
+
+    private getCaptionsMode = new Callable("getCaptionsMode", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("On");
+        },
+    });
+
+    private setCaptionsMode = new Callable("setCaptionsMode", {
+        signature: {
+            args: [new StdlibArgument("mode", ValueKind.String)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (_interpreter) => {
+            return BrsBoolean.True;
+        },
+    });
+
+    private getCaptionsOption = new Callable("getCaptionsOption", {
+        signature: {
+            args: [new StdlibArgument("option", ValueKind.String)],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("Default");
+        },
+    });
+
+    private getClockFormat = new Callable("getClockFormat", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("12h");
+        },
+    });
+
+    private enableAppFocusEvent = new Callable("enableAppFocusEvent", {
```

Comment:
Similar to the captions mode, let's keep track of this value in a variable.

Reply:
The way I am understanding this is creating an instance variable `private enableAppFocus = BrsBoolean.False`, create the variable's getter/setter functions. Next, when the consumer of this method passes a boolean argument, true, for example, then inside `impl` we should set the variable (using the setter function) ? Is my line of thought correct or far off? :)

The same thought process also extends to other functions that take arguments as well.

Reply:
You're pretty close, except that we don't need an explicit getter or setter function like you've created — just a simple `this.enableAppFocusStatus = enable; return this.enableAppFocusState;` (or as a oneliner, `return (this.enableAppFocusStatus = enable;`) is enough!  Because those variables are private to this instance of `roDeviceInfo`, no other TypeScript code outside of this class will be able to access them.  Your getters and setters expose that `this.enableAppFocusStatus` property to other TypeScript contexts, but don't appear to be called anywhere!